bugwarrior/services/deck.py

```diff
@@ -0,0 +1,192 @@
+# * Cards will be mapped to tasks
+# * Labels will be mapped to tags
+class NextcloudDeckClient(ServiceClient):
+    def __init__(self, base_uri, username, password):
+        self.base_uri = base_uri
+        self.username = username
+        self.password = password
+        self.api_base_path = 'index.php/apps/deck/api/v1.0'
+
+        self.session = requests.session()
+        self.session.auth = (self.username, self.password)
+        self.session.headers.update({
+            'Accept': 'application/json',
+            'OCS-APIRequest': 'true',
+        })
+
+    # see https://deck.readthedocs.io/en/latest/API/#boards for API docs
+    def get_boards(self):
+        response = self.session.get(f'{self.base_uri}/{self.api_base_path}/boards')
+        return response.json()
+
+    # see https://deck.readthedocs.io/en/latest/API/#stacks for API docs
+    def get_stacks(self, board_id):
+        response = self.session.get(
+            f'{self.base_uri}/{self.api_base_path}/boards/{board_id}/stacks'
+        )
+        return response.json()
+
+
+class NextcloudDeckIssue(Issue):
+    AUTHOR = 'nextclouddeckauthor'
+    BOARD_ID = 'nextclouddeckboardid'
+    BOARD_TITLE = 'nextclouddeckboardtitle'
+    STACK_ID = 'nextclouddeckstackid'
+    STACK_TITLE = 'nextclouddeckstacktitle'
+    CARD_ID = 'nextclouddeckcardid'
+    TITLE = 'nextclouddeckcardtitle'
+    DESCRIPTION = 'nextclouddeckdescription'
+    ORDER = 'nextclouddeckorder'
+    ASSIGNEE = 'nextclouddeckassignee'
+
+    UDAS = {
+        AUTHOR: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Issue Author',
+        },
+        BOARD_ID: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Board ID',
+        },
+        BOARD_TITLE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Board Title',
+        },
+        STACK_ID: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Stack ID',
+        },
+        STACK_TITLE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Stack Title',
+        },
+        CARD_ID: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Card ID',
+        },
+        TITLE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Card Title',
+        },
+        DESCRIPTION: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Card Description',
+        },
+        ORDER: {
+            'type': 'numeric',
+            'label': 'Nextcloud Deck Order',
+        },
+        ASSIGNEE: {
+            'type': 'string',
+            'label': 'Nextcloud Deck Assignee(s)',
+        }
+    }
+
+    UNIQUE_KEY = (BOARD_ID, STACK_ID, CARD_ID,)
+
+    def to_taskwarrior(self):
+        return {
+            'project': self.extra['board']['title'].lower().replace(' ', '_'),
+            'priority': self.get_priority(),
+            # 'annotations': self.record.get('annotations', []),
```

Comment:
Were you planning to complete annotations support? I believe every other service currently has this feature.

Reply:
I did not initially, but I just added it. It does slow down the process significantly, though, because there is a specific call needed per card / issue. For my cases it's about factor 10 (20 seconds instead of 2). As far as I can see it, it's not possible to skip annotation importing via configuration somehow, correct? How would you handle that - it annoys me quite a lot if it slows down the process considerably, but I also see value in completeness.

Reply:
This is typically handled by setting the main configuration option "annotation_comments" to `False`. When this is set `IssueService.build_annotations` just throws the annotations away, so several services have implemented logic to avoid calling the `client.get_comments` method when when this is the case. (You can just pass an empty list to `build_annotations`. You can see the github and gitlab services for examples.)

I don't believe we have any way to disable annotations on an individual service basis or that anyone has requested that.